### ISSUE TITLE
Mypy infers types of NamedTuple subclass as 'nothing' when using isinstance.

### ISSUE THREAD

**Issue by toshihikoyanase** [python/mypy#5640]
I encountered an issue that `mypy` failed to infer a type of a variable. This issue occurs when `isinstance()` is applied to a subclass of `NamedTuple`.

Environment:
- Python 3.6.5
- mypy: 0.630

Code
```python
from typing import NamedTuple


class Base:
    pass


class BaseTuple(NamedTuple):
    value: float


class MyTuple(BaseTuple, Base):
    pass


def f(o: Base) -> float:
    if isinstance(o, MyTuple):
        return o.value + 10
    return 0


t = MyTuple(1)
print(f(t))
```

Python 3.6.5 successfully executes this code and returns `11` as expected. But `mypy` 0.630 fails to infer the type of `o` after checking its type using `isinstance` as follows:

```
sample.py:18: error: <nothing> has no attribute "value"
```

Observations:
- Current master head (commit 5751f42ff79265deb4126075b2511b823d006ae0) also has this issue.
- `mypy` 0.620 successfully infer the type and does not show this error message.
- This issue seems to occur from https://github.com/python/mypy/commit/3fb16a2b8c5439074e39b75feebc109a439eede3.
- `mypy` 0.630 successfully infer the type if `BaseTuple` is not a subclass of NamedTuple as follows:
```python
class BaseTuple:

    def __init__(self, value: float) -> None:
        self.value = value
```

**Comment by ilevkivskyi** [comment]
Thanks for reporting! We are sorry for this regression. I will try fixing this shortly.

**Comment by toshihikoyanase** [comment]
This fix solved actual problems in my project. Thank you for your swift action!
